Makefile.Common

```diff
@@ -224,3 +224,8 @@ moddownload:
 gci: $(TOOLS_BIN_DIR)/gci
 	@echo "running $(GCI)"
 	@$(GCI) write -s standard -s default -s "prefix(github.com/open-telemetry/opentelemetry-collector-contrib)" $(ALL_SRC_AND_DOC)
+
+.PHONY: generate
+generate: $(TOOLS_BIN_DIR)/mdatagen
```

Comment:
Sometimes make escaped me, but I'm not following why `$(TOOLS_BIN_DIR)/mdatagen` was added here.

Reply:
That is used to ensure that the `mdatagen` binary/file is present before running the `generate` target. And as `$(TOOLS_BIN_DIR)/mdatagen` is a [make target](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34553/files/c453592b65d96a75263ba3c19cf6b33c0bdd1380#diff-42d381ee1180be37a0b2dda045820c2550ec5cffcdf35b2423e9ccff01b5dfc1R56), it will install it otherwise.